src/PrestaShopBundle/Resources/config/services/bundle/controller.yml

```diff
@@ -2,29 +2,35 @@ services:
   _defaults:
     public: true
 
+  _instanceof:
+    # cannot use autocnfigure under _instanceof so we need to define it here
+    PrestaShopBundle\Controller\Admin\PrestaShopAdminController:
+      autowire: true
+      public: true
+      tags:
+        - 'controller.service_arguments'
+        - 'container.service_subscriber'
+    PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController:
+      calls:
+        - [ setContainer, [ '@service_container' ] ]
+
+
+  # Legacy controllers don't extend the base controller, so we need to inject the container manually.
   PrestaShopBundle\Controller\Api\ApiController:
     class: PrestaShopBundle\Controller\Api\ApiController
     calls:
       - [ setLogger, [ '@logger' ] ]
       - [ setContainer, [ '@service_container' ] ]
-    tags:
-      - { name: !php/const PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController::PRESTASHOP_CORE_CONTROLLERS_TAG }
```

Comment:
The tag is not needed anymore? Then it means the `ContainerInjectionPass` doesn't find tagged services anymore since most of them don't have the tag anymore right?